### PR TITLE
Simplify setup wizard 2

### PR DESCRIPTION
**Pull request by hws47a**
### Description (*)

Simplify setup wizard code:
- remove language configuration as it's unused
- replace custom LazyControllerAbstractFactory with Laminas ReflectionBasedAbstractFactory
- remove all unused modules from laminas mvc app
- remove unused dependencies previously used in LazyControllerAbstractFactory

Note:
- laminas-filter is not used directly in the code, but it's used by laminas-captcha and it's not required (but suggested) by the package.

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/528

### Fixed Issues (if relevant)

1. It also fixes magento/magento2#32442 and replaces another PR made for it

### Manual testing scenarios (*)
Check that `/setup/` still works

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by hws47a** [comment]
@magento run all tests

**Comment by hws47a** [comment]
@magento run all tests

**Comment by hws47a** [comment]
@magento run Functional Tests CE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests CE

**Comment by ihor-sviziev** [comment]
@hws47a, all tests are passing now!
I believe this PR replaces https://github.com/magento/magento2/pull/32475 and https://github.com/magento/magento2/pull/32512. Is that correct?

**Comment by hws47a** [comment]
@ihor-sviziev,  previous was simplifying controllers and removing angular.

This one is removing "lazy controller" and unnecessary dependencies

**Comment by hws47a** [comment]
@magento run all tests

**Comment by hws47a** [comment]
@ihor-sviziev it actually replaces #32449

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests CE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests CE

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests looks not related to changes from this PR.

File: composer.json [hunk 1]
```diff
@@ -43,19 +43,17 @@
         "laminas/laminas-di": "^3.2.0",
         "laminas/laminas-eventmanager": "^3.0.0",
         "laminas/laminas-feed": "^2.13.0",
-        "laminas/laminas-form": "^2.10.0",
+        "laminas/laminas-filter": "^2.11",
         "laminas/laminas-http": "^2.6.0",
         "laminas/laminas-i18n": "^2.7.3",
         "laminas/laminas-json": "^3.2.0",
-        "laminas/laminas-log": "^2.13",
```
Comment (mpchadwick) [inline composer.json:50]: Why was `laminas/laminas-log` removed?

**Pull request merged**